backend/src/resolvers/post.resolver.ts

```diff
@@ -14,11 +14,11 @@ export const PostMutationResolver: Pick<MutationResolvers.Type
     | 'upvote'
     > = {
     // @ts-ignore
-    write: (parent, { post }, { dataSources }) => {
-        return dataSources.databaseAPI.writePost(post);
+    write: (parent, { post }, ctx) => {
+        return ctx.dataSources.databaseAPI.writePost(post, ctx.userId);
```

Comment:
If you use the `initialize()` method of a `dataSource`, you could set the context there instead of passing it from the resolvers.